README.rst

```diff
@@ -15,104 +15,171 @@ sqlitedict -- persistent ``dict``, backed-up by SQLite and pickle
 A lightweight wrapper around Python's sqlite3 database with a simple, Pythonic
 dict-like interface and support for multi-thread access:
 
+Usage
+=====
+
+Write
+-----
+
 .. code-block:: python
 
-  >>> from sqlitedict import SqliteDict
-  >>> mydict = SqliteDict('./my_db.sqlite', autocommit=True)
-  >>> mydict['some_key'] = any_picklable_object
-  >>> print mydict['some_key']  # prints the new value
-  >>> for key, value in mydict.iteritems():
-  >>>     print key, value
-  >>> print len(mydict) # etc... all dict functions work
-  >>> mydict.close()
+    from sqlitedict import SqliteDict
+    db = SqliteDict("./db.sqlite")
+
+    db["1"] = {"name": "first item"}
+    db["2"] = {"name": "second item"}
+    db["3"] = {"name": "yet another item"}
+
+    # commit to save the objects.
+    db.commit()
 
-Pickle is used internally to (de)serialize the values. Keys are arbitrary strings,
-values arbitrary pickle-able objects.
+    db["4"] = {"name": "yet another item"}
+    # oops, forgot to commit here, that object will never be saved.
 
-If you don't use autocommit (default is no autocommit for performance), then
-don't forget to call ``mydict.commit()`` when done with a transaction:
+    # always remember to commit, or enable autocommit with SqliteDict("./db.sqlite", autocommit=True)
+    # autocommit is off by default for performance.
+
+Read
+----
 
 .. code-block:: python
 
-  >>> # using SqliteDict as context manager works too (RECOMMENDED)
-  >>> with SqliteDict('./my_db.sqlite') as mydict:  # note no autocommit=True
-  ...     mydict['some_key'] = u"first value"
-  ...     mydict['another_key'] = range(10)
-  ...     mydict.commit()
-  ...     mydict['some_key'] = u"new value"
-  ...     # no explicit commit here
-  >>> with SqliteDict('./my_db.sqlite') as mydict:  # re-open the same DB
-  ...     print mydict['some_key']  # outputs 'first value', not 'new value'
+    from sqlitedict import SqliteDict
+    db = SqliteDict("./db.sqlite")
 
+    print("There are {} items in the database".format(len(db)))
```

Comment:
Please use f-strings (if we support Python 3.6+, I'm sure – we probably should!). Or else normal `%s` if not. We avoid `format()` in our codebase.